### DISCUSSION TITLE
Could the MsSql Module be extended to allow WithDatabase please?

### DISCUSSION THREAD

**Discussion by PureKrome** [testcontainers/testcontainers-dotnet#1112] [General]
the default settings for the MsSql module is that is uses the `Master` db for all my own DB stuff.

taken from this code snippet

```
        return base.Init().WithImage("mcr.microsoft.com/mssql/server:2019-CU18-ubuntu-20.04").WithPortBinding(1433, assignRandomHostPort: true)
            .WithEnvironment("ACCEPT_EULA", "Y")
            .WithDatabase("master")
            .WithUsername("sa")
            .WithPassword("yourStrong(!)Password")
            .WithWaitStrategy(Wait.ForUnixContainer().AddCustomWaitStrategy(new WaitUntil()));
```

Would be nice if we can just have the option to set that value please? Why? When i'm testing a single TEST i somemtimes put a break point and then to _manually_ check some data in the db.

So it would be really nice to have my db separate from `master`.

This way, we could also _potenetlly_ have a unique DB per test, in the same Container (versus, spawning new conainters per test - `.WithDatabase($"Testing-{Guid.NewGuid()}")`. RavenDb has been doing this since day 1, which works nice!

**Comment by 4dri4ng** [comment]
Hi :)
you can use something like that:
```
            services.AddDbContext<YourMagicContext>(options =>
                options.UseSqlServer(new SqlConnectionStringBuilder(_dbContainer.GetConnectionString())
                {
                    InitialCatalog = "YourMagicDbName"
                }.ToString()));
```
you will override 'master' in connection string :)

**Reply by PureKrome** [reply]
Oh wow - yep! this is great! Would love to have the documentation updated to include this trick. Works really well.
